tests/labs/lie/functional/common.py

```diff
@@ -56,57 +57,81 @@ def get_test_cfg(op_name, dtype, dim, data_shape, module=None):
 
 # Sample inputs with the desired types.
 # Input type is one of:
-#   ("tangent", dim)  # torch.rand(batch_size, dim)
-#   ("group", module) # e.g., module.rand(batch_size)
+#   ("tangent", dim)  # torch.rand(*batch_size, dim)
+#   ("group", module) # e.g., module.rand(*batch_size)
 #   ("quat", dim)     # sampled like tangent but normalized
-#   ("matrix", shape) # torch.rand((batch_size,) + shape)
-def sample_inputs(input_types, batch_size, dtype, rng, module=None):
+#   ("matrix", shape) # torch.rand((*batch_size,) + shape)
+#
+# `batch_size` can be a tuple.
+def sample_inputs(input_types, batch_size, dtype, rng):
+    if isinstance(batch_size, int):
+        batch_size = (batch_size,)
+
     def _sample(input_type):
         type_str, param = input_type
 
         def _quat_sample():
-            q = torch.rand(batch_size, param, dtype=dtype, generator=rng)
-            return q / torch.norm(q, dim=1, keepdim=True)
+            q = torch.rand(*batch_size, param, dtype=dtype, generator=rng)
+            return q / torch.norm(q, dim=-1, keepdim=True)
 
         sample_fns = {
             "tangent": lambda: torch.rand(
-                batch_size, param, dtype=dtype, generator=rng
+                *batch_size, param, dtype=dtype, generator=rng
             ),
-            "group": lambda: param.rand(batch_size, generator=rng, dtype=dtype),
+            "group": lambda: param.rand(*batch_size, generator=rng, dtype=dtype),
             "quat": lambda: _quat_sample(),
             "matrix": lambda: torch.rand(
-                (batch_size,) + param, generator=rng, dtype=dtype
+                (*batch_size,) + param, generator=rng, dtype=dtype
             ),
         }
         return sample_fns[type_str]()
 
     return tuple(_sample(type_str) for type_str in input_types)
 
 
-# Run the test for a Lie group operator
+# Run some unit tests for a Lie group operator:
+# checks:
+#   - jacobian of default torch autograd consistent with custom backward implementation
+#   - multi-batch output consistent with single-batch output
 def run_test_op(op_name, batch_size, dtype, rng, dim, data_shape, module):
+    is_multi_batch = not isinstance(batch_size, int)
+    bs = len(batch_size) if is_multi_batch else 1
     all_input_types, atol = get_test_cfg(op_name, dtype, dim, data_shape, module=module)
     for input_types in all_input_types:
         inputs = sample_inputs(input_types, batch_size, dtype, rng)
         funcs = (
-            tuple(left_project_func(module, x) for x in inputs)
+            tuple(left_project_func(module, x, bs) for x in inputs)
```

Comment:
Wondering if we need to check `bs=None` or `dim_out=None`, which assumes no broadcasting and is most common user-case.

Reply:
I thought the case `tuple()` covered this for the no batch size (see line 10). Or do you mean something else? 

The tests for different `dim_out` are in #527.